superset/views/core.py

```diff
@@ -3080,50 +3080,100 @@ def geocoding(self):
+        table_name = request.json.get("tableName", "")
+        error_message = "No columns found for table with name {0}".format(table_name)
+        try:
+            columns = reflection.Inspector.from_engine(db.engine).get_columns(
+                table_name
             )
-
+            if columns:
+                column_names = [column["name"] for column in columns]
+            else:
+                return json_error_response(error_message, status=400)
+        except:
+            return json_error_response(error_message, status=400)
         return json.dumps(column_names)
 
     @api
     @has_access_api
     @expose("/geocoding/geocode", methods=["POST"])
     def geocode(self) -> Response:
-        # TODO this has to be removed and replaced with a call to the get_data_from_table method
+        table_name = request.json.get("datasource", "")
+        columns = [
+            request.json.get("streetColumn"),
+            request.json.get("cityColumn"),
+            request.json.get("countryColumn"),
+        ]
+        lat_column = request.json.get("latitudeColumnName", "lat")
+        lon_column = request.json.get("longitudeColumnName", "lon")
+        override_if_exist = request.json.get("overwriteIfExists", False)
+        save_on_stop_geocodeing = request.json.get("saveOnErrorOrInterrupt", True)
+        data = [()]
+
         try:
-            dat = self._geocode(request.data)
-            # TODO enrichted data has to be written into database
-            return json_success(json.dumps(dat))
+            if not override_if_exist and self._does_column_name_exist(
+                table_name, lat_column
+            ):
+                raise ValueError(
+                    "Column name {0} for latitude is already in use".format(lat_column)
+                )
+            if not override_if_exist and self._does_column_name_exist(
+                table_name, lon_column
+            ):
+                raise ValueError(
+                    "Column name {0} for longitude is already in use".format(lon_column)
+                )
+
+            data = self._load_data_from_columns(table_name, columns)
+        except ValueError as e:
+            return json_error_response(e.args[0], status=400)
         except Exception as e:
-            return json_error_response(e.args)
+            return json_error_response(e.args[0], status=500)
 
-    def _get_editable_tables(self):
-        """ Get tables which are allowed to create columns (allow dml on their database) """
-        tables = []
-        for database in (
-            db.session.query(models.Database).filter_by(allow_dml=True).all()
-        ):
-            for table in (
-                db.session.query(SqlaTable).filter_by(database_id=database.id).all()
-            ):
-                tables.append(models.TableDto(table.id, table.name, table.database_id))
-        return tables
+        try:
+            data = self._geocode(data)
```

Comment:
_geocode, or rather the method which actually geocodes currently expects a dict, change that method to accept and work with a list